src/lib.rs

```diff
@@ -342,6 +350,10 @@ fn apply_filter_with_flags(bpf_filter: BpfProgramRef, flags: libc::c_ulong) -> R
         return Err(Error::EmptyFilter);
     }
 
+    if bpf_filter.len() > u16::MAX as usize {
+        return Err(Error::TooManyInstructions);
+    }
```

Comment:
I'd rather use `bpf_filter.len().try_into::<u16>().map_err(...)?` instead. Could be inlined into the `bfp_prog` declaration below.

Reply:
Makes sense! I moved the `bfp_prog` declaration, so `PR_SET_NO_NEW_PRIVS` is only called if we know that the the arguments are sane.